### ISSUE TITLE
premake group for vendor projects

### ISSUE THREAD

**Issue by ZebNoid** [TheCherno/Hazel#48]
Hello.

Number of projects in vendor folder are incrising. How about separating projects in subfolders. Using [premake group](https://github.com/premake/premake-core/wiki/group).

In general something like this:
```
-- premake5.lua
...
group "Vendor"
include "Hazel/vendor/GLFW"
include "Hazel/vendor/Glad"
include "Hazel/vendor/imgui"

group "Engine"
project "Hazel"
...
```

**Comment by MrBrN197** [comment]
I don't think this is an issue

**Comment by ZebNoid** [comment]
Yes it's not an issue, it's just suggestion

**Comment by Gaztin** [comment]
@MrBrN197 GitHub Issues are not always used for problems. Suggestions are common.

Personally, I like this suggestion. This is how I deal with my dependencies in my projects. I usually add a "Third Party" group where I shove the external libraries. It helps with readability and getting a better picture of the workspace itself.

**Comment by SeppahBaws** [comment]
I would like to mention that I already made a pullrequest about a month ago, where I implemented this: #19 
I kept it fairly simple and just grouped the dependencies together, but you can go further than that.
e.g. there could be a group for engine projects (if there would be more than one) and one for the sandbox projects, but that's for Cherno to decide.

**Comment by Gaztin** [comment]
Closing since this has been merged since.
